lib/chitragupta/util.js

```diff
@@ -39,6 +40,10 @@ function populateServerData(dataParam) {
   }
   data.data.request.params = data.data.request.params.substring(0, fieldLimits.PARAMS) || '';
 
+  data.data.request.headers = data.data.request.headers || headers;
+
+  data.data.request.headers = data.data.request.headers.substring(0, fieldLimits.HEADERS) || '';
```

Comment:
this will break if someone is passing headers in their object as dict because,
```
data.data.request.headers = data.data.request.headers || headers;
```
This will keep `data.data.request.headers` as dict and in below line it will take substring of a dict and thus break